tests/unit/mixins/pagination-support-serializer-test.js

```diff
@@ -0,0 +1,48 @@
+import PaginationSupportSerializerMixin from '@bagaar/ember-data-bagaaravel/mixins/pagination-support-serializer';
+import JSONAPIBagaaravelSerializer from '@bagaar/ember-data-bagaaravel/serializers/json-api-bagaaravel';
+import JSONAPIAdapter from 'ember-data/adapters/json-api';
+import { setupTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+
+module('Unit | Mixin | pagination-support-serializer', function (hooks) {
+  setupTest(hooks);
+
+  test('it creates pagination meta', async function (assert) {
+    let Adapter = JSONAPIAdapter.extend({
+      ajax() {
+        return Promise.resolve({
+          data: [],
+          links: {
+            first: 'https://api.com/users?page=1&per_page=15',
+            last: 'https://api.com/users?page=9&per_page=15',
+            next: 'https://api.com/users?page=2&per_page=15',
+            previous: null,
+            self: 'https://api.com/users?page=1&per_page=15',
+          },
+          meta: {
+            perPage: 15,
+            total: 125,
+          },
+        });
+      },
+    });
+
+    let Serializer = JSONAPIBagaaravelSerializer.extend(PaginationSupportSerializerMixin);
+
+    this.owner.register('adapter:application', Adapter);
```

Comment:
You could also generate an application adapter in our dummy app and that would automagically be used in our tests. You could use `reopen` to adapt behaviour per test. But this approach seemed much cleaner/explicit though it's a bit more verbose.

Reply:
I think this is a way better solution anyway. This code is very specific for this test. It shouldn't be added to the serializer that will be used by other tests. Good job!